mp_add_d.c

```diff
@@ -10,6 +10,15 @@ mp_err mp_add_d(const mp_int *a, mp_digit b, mp_int *c)
    int ix, oldused;
    mp_digit *tmpa, *tmpc;
 
+   /* fast path for a == c */
+   if (a == c &&
+       !mp_iszero(c) &&
+       c->sign == MP_ZPOS &&
+       c->dp[0] + b < MP_DIGIT_MAX) {
```

Comment:
I think parentheses are mandatory now.

i would move the `mp_iszero` check into the block and add something like e.g.: `c->used += mp_iszero(c)` but that's just me.

Reply:
I reworked it a bit and added the parens. However I didn't to this c->used += mp_iszero(c) thing and the more complicated equivalent for subtraction to avoid adding too much code. I mean it is the fast path, it is sufficient if it handles the common cases.